### DISCUSSION TITLE
Getting very distorted audio output

### DISCUSSION THREAD

**Discussion by anujsahani01** [coqui-ai/TTS#2794] [General Q&A]
Dataset:
Prepared a dataset of Indian actress consisting 212 audio samples.
Dataset can be downloaded from [here](https://drive.google.com/drive/folders/1FSSz1yrBzp7D88MD1KD_kv-4tV-b7h9Y?usp=sharing)
Removed the noise from the data using [rnnoise](https://github.com/xiph/rnnoise) transcribed the audio using [whisper](https://github.com/openai/whisper)

**Audio after noise reduction**

https://github.com/coqui-ai/TTS/assets/83875986/00668cf5-718e-4e9c-bc36-d6e3fbc286ad

Made a custom formatter (for reading my own dataset) in file  ``` TTS/tts/datasets/formatters.py ``` 
```
def custom_formatter(root_path, meta_file, **kwargs):
    df  = pd.read_csv(meta_file)
    items = []
    speaker_name = "custom_formatter"
    for line in df['inp']:
        cols = line.split("|")
        wav_file = os.path.join(root_path, cols[0].replace(" ", "") + ".wav")
        text = cols[2].replace(/[!@#$%^&*]/g, "")
        items.append({"text":text, "audio_file":wav_file, "speaker_name":speaker_name, "root_path": root_path})
    return items
```

Tried the following combination of  hyper parameters but none of them gave good results
```
    batch_size=(16 | 1 | 2 | 4),
    eval_batch_size=(16 | 1 | 2 | 4),
    num_loader_workers=(4 | 8),
    num_eval_loader_workers=(4 | 8),
    run_eval=True,
    test_delay_epochs=-1,
    epochs=(200 | 250 | 300)
    text_cleaner="phoneme_cleaners",
    phoneme_language="en-us",
    print_step=1000,
    lr=(0.0005 | 0.001)
    use_phonemes=False,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    save_step=7000,
```
## Observations: 
* **num_loader_workers** on increasing its value model performance decreased.
*  **batch_size** no significant change in model performance for both the values.
* **epochs** for larger epoch model started overfitting.
* **lr** no significant change in model performance for both the values.

### This is the output audio that my model generated 

https://github.com/coqui-ai/TTS/assets/83875986/e32ef888-e9c7-4a26-9b23-64a1d5d993ba

For more reference please go through my  [git repo](https://github.com/anujsahani01/VoiceCloning-coqui-TTS)
Give me some suggestion on how can i improve my models performance and also correct me if i am wrong with my concepts.

I went through other discussion but none of the ways suggested there gave me good results.
@erogol  please take out some time and help me out with this.
Any of your inputs will be highly appreciated.
Thank You !

**Comment by anujsahani01** [comment]
Anyone please help me out.
Any of your inputs will be highly appreciated.
Thank You !

**Reply by pivolan** [reply]
I am sure that problem is in your dsataset and metadata.csv file.
I also tried train vits model zero, not finetuning, and get this results for german language on 250 audio files:
https://static.bottalk.work/dan_250.wav

what model did you use?
Try to use vits model, enable phoneme: use_phonemes=True
You should listen recognizible speech on 100 epochs already, if still not - there are some error.
LR change will not help, default value is ok.

this has no affect on the quality:
num_loader_workers on increasing its value model performance decreased.
batch_size no significant change in model performance for both the values.

**Comment by pivolan** [comment]
> 
Also your dataset not good, there are different speed of pronounce, and too fast speech.
Also try to convert all audios to 1 channel, because your audios has 2 channels.

**Comment by pivolan** [comment]
and check this: 
audio_config = VitsAudioConfig(
    sample_rate=22050, win_length=1024, hop_length=256, num_mels=80, mel_fmin=0, mel_fmax=None
)
this is my  settings for audio, it is just default values, and I've never do any experiments with changing this params, results are always good.
